### ISSUE TITLE
为什么我训练得到的iou这么低

### ISSUE THREAD

**Issue by NYB-AAA** [tsinghua-fib-lab/UV-SAM#7]
![image](https://github.com/tsinghua-fib-lab/UV-SAM/assets/130569364/21547461-773e-4f22-9ea9-cb56370fdcdb)
用的是一张4090，在北京数据集上训练的，预训练权重用的是segmentation源码里的mit_b0.pth权重，求解答，UV部分的iou只有个位数，这是为什么

**Comment by lianzi666** [comment]
我的勉强两位数，测试iou=0,请问你解决了吗？

**Comment by NYB-AAA** [comment]
我跑了四次，现象是一开始UV的iou很低，之后就会变成两位数，但是4次的结果都无法达到论文里的iou精度，其他三个指标p,r f1也无法达到精度，只有一个指标勉强接近，但是我的测试是可以正常进行的，只要训练完把checkpoints路径改了就可以测试，论文里说lr,weight decay, λ是在{0.005, 0.0005,0.00005}, {0.01, 0.001} and {0.1, 1, 10}里网格搜索的，代码里默认是0.005，0.001，0.1，我现在考虑的是不是得这三个参数排列组合才能得到论文的结果，但我是租服务器的，排列组合有点伤不起

**Comment by Ztltl** [comment]
遇到了同样的问题

**Comment by Xin-Zh00** [comment]
Our model was trained on a single RTX-4090. We used the grid search algorithm to search for parameters.



> 我跑了四次，现象是一开始UV的iou很低，之后就会变成两位数，但是4次的结果都无法达到论文里的iou精度，其他三个指标p,r f1也无法达到精度，只有一个指标勉强接近，但是我的测试是可以正常进行的，只要训练完把checkpoints路径改了就可以测试，论文里说lr,weight decay, λ是在{0.005, 0.0005,0.00005}, {0.01, 0.001} and {0.1, 1, 10}里网格搜索的，代码里默认是0.005，0.001，0.1，我现在考虑的是不是得这三个参数排列组合才能得到论文的结果，但我是租服务器的，排列组合有点伤不起

**Comment by NYB-AAA** [comment]
你好，可以提供一下网格搜索的代码吗

**Comment by Ztltl** [comment]
> 你好，可以提供一下网格搜索的代码吗？

我排列组合依次试的，3*3*2=18种，要试好久

**Comment by NYB-AAA** [comment]
> > 你好，可以提供一下网格搜索的代码吗？
> 
> 我排列组合依次试的，3_3_2=18种，要试好久

达到论文里的精度了吗，是什么参数能达到精度 啊，还有seed设置，因为不可能是默认的1234吧

**Comment by Ztltl** [comment]
> > > 你好，可以提供一下网格搜索的代码吗？
> > 
> > 
> > 我排列组合依次试的，3_3_2=18种，要试好久
> 
> 达到论文里的精度了吗，是什么参数能达到精度 啊，还有seed设置，因为不可能是默认的1234吧

还没跑完 目前没达到 不懂啊

**Comment by 1064028322** [comment]
遇到了同样的问题

**Comment by Xin-Zh00** [comment]
> ![image](https://private-user-images.githubusercontent.com/130569364/313975862-21547461-773e-4f22-9ea9-cb56370fdcdb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4NDkxNDAsIm5iZiI6MTcxMjg0ODg0MCwicGF0aCI6Ii8xMzA1NjkzNjQvMzEzOTc1ODYyLTIxNTQ3NDYxLTc3M2UtNGYyMi05ZWE5LWNiNTYzNzBmZGNkYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxMVQxNTIwNDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZDllODQyM2I3ZTQzMzVlMDAxZjhmNmU0YjJiNWU5YTRiNWYwMTQyYWMwNTY3NzZmODMwZmJlNDA4ZWM1MGZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ErUiiwz0xUXogK6KxSTu2qu-XvJ3RKVtNxq8ULihtOg) 用的是一张4090，在北京数据集上训练的，预训练权重用的是segmentation源码里的mit_b0.pth权重，求解答，UV部分的iou只有个位数，这是为什么

It seems that your pretrained model was not loaded correctly. Please check the path of the pretrained model and the key of the model. The pretrained model's parameters are the same as the mmsegmentation model. The difference between mmsegmentation and pretrained model is that we just changed the key of pretrained model. The pretrained model can be downloaded from the following link: *[Google Drive](https://drive.google.com/drive/folders/1szict970v0Z7LrY78jPQkNG1UzTHT1gb?usp=sharing)*.

**Comment by Xin-Zh00** [comment]
> 你好，可以提供一下网格搜索的代码吗

I have provided the grid search file in the **./grid_search.sh** path, which can be used to search the best hyperparameters for the model.

**Comment by Xin-Zh00** [comment]
> > > 你好，可以提供一下网格搜索的代码吗？
> > 
> > 
> > 我排列组合依次试的，3_3_2=18种，要试好久
> 
> 达到论文里的精度了吗，是什么参数能达到精度 啊，还有seed设置，因为不可能是默认的1234吧

In my reproduced experiments, for Beijing dataset, learning rate is 0.0005, weight decay is 0.01, and SAM_weights is 0.1. For Xi'an dataset, learning rate is 0.0005, weight decay is 0.001, and SAM_weights is 1. The seed is fixed to 1234 for reproducibility.

**Comment by Xin-Zh00** [comment]
> 我的勉强两位数，测试iou=0,请问你解决了吗？

I have updated the code and the results.
For Beijing dataset, I used the SegSAMPLerEmbMLP_config.py to train the model.
For Xi'an dataset, I used the SegSAMPLerEmbAdd_config.py to train the model.
